src/data/Path.hpp

```diff
@@ -44,69 +45,62 @@ class Path
  # If root was set to /some/root, then the final path would be:
  <grist>/some/root/path/to/basename.suffix(member)
  \endverbatim
+ *
+ * For performance, this class uses string views to decompose and manipulate an
+ * existing string. Path::Parts must not outlive the scope of any strings it is
+ * passed.
  */
 class Path::Parts
 {
   public:
 	Parts() {}
-	Parts(const StringPart& path) { SetTo(path); }
+	Parts(std::string_view path);
 
-	void SetTo(const StringPart& path);
-	void GetPath(StringBuffer& buffer, const behavior::Behavior& behavior)
-		const;
-	inline String ToPath(const behavior::Behavior& behavior) const;
+	std::string ToPath(const behavior::Behavior& behavior) const;
 
 	bool IsAbsolute() const;
 
-	const StringPart& Grist() const { return fGrist; }
-	void SetGrist(const StringPart& grist) { fGrist = grist; }
-	void UnsetGrist() { fGrist.Unset(); }
+	std::string Grist() const { return std::string{fGrist}; }
```

Comment:
My idea was to remove the lifetime requirement so that returned values could outlive the class, but perhaps I should make it the user's responsibility to create a string from the view?

Reply:
I'm fine either way...